---
title: "Bernd Klaus - Talks, Publications and Software"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
---


<!--

<p>
<a href="http://www.embl.de"><img src="klaus.jpeg" width=100 align="left"></a>
<a href="http://www.embl.de"><img src="embl.jpg" width=300 align="right"></a>
</p>
<br>
<br>
<br>
<br>
<br>
<br>
-->

## Talks and Notes 

Here I gather various notes and talks that might be of interest.


* Tools for single cell RNA-Seq --- slides (June 2018): <http://b-klaus.de/slides_sc.pdf>
* Tools for single cell RNA-Seq (June 2018): <http://b-klaus.de/sc_tools.html>
* MDS vs t--SNE (October 2017): <http://b-klaus.de/Multi_Dimensional_Scaling_vs_t_SNE.pdf>

## Publications

### 2018

- **Gain of CTCF-Anchored Chromatin Loops Marks the Exit from Naive Pluripotency** --- Aleksandra Pękowska, **Bernd Klaus**, Wanqing Xiang, Jacqueline Severino, Nathalie Daigle, Felix A. Klein, Małgorzata Oleś, Rafael Casellas, Jan Ellenberg, Lars M. Steinmetz, Paul Bertone, Wolfgang Huber --- [Cell Syst](http://dx.doi.org/10.1016/j.cels.2018.09.003)

- **Machine Learning Predicts the Yeast Metabolome from the Quantitative Proteome of Kinase Knockouts** --- Zelezniak A, Vowinckel J, Capuano F, Messner CB, Demichev V, Polowsky N, Mülleder M, Kamrad S, **Klaus B**, Keller MA, Ralser M. --- [Cell Syst 7(3)](http://dx.doi.org/10.1016/j.cels.2018.08.001)

- **An end to end workflow for differential gene expression using Affymetrix microarrays**  --- **Bernd Klaus** and **Stefanie Reisenauer** --- version 2 , 2018 --- [F1000Research 2018, 5:1384](http://dx.doi.org/10.12688/f1000research.8967.2)


### 2017

- **Landscape of nuclear transport receptor cargo specificity.** Mackmull MT, **Klaus B**, Heinze I, Chokkalingam M, Beyer A, Russell RB, Ori A, Beck M.  [Mol Syst Biol. 2017 Dec;13(12) 962.](http://dx.doi.org/10.15252/msb.20177608)

- **voomDDA: discovery of diagnostic biomarkers and classification of RNA-seq data**
Zararsiz G, Goksuluk D, **Klaus B**, Korkmaz S, Eldem V, Karabulut E, Ozturk A. (2017)
--- [PeerJ 5](http://dx.doi.org/10.7717/peerj.3890) --- [voomDDA webtool]( http://www.biosoft.hacettepe.edu.tr/voomDDA/)

- **Metabolic shifts in residual breast cancer drive tumor recurrence** --- Havas, Kristina M; Milchevskaya, Vladislava; Radic, Ksenija; Alladin, Ashna; Kafkia, Eleni; Garcia, Marta; Stolte, Jens; **Klaus, Bernd**; Rotmensz, Nicole; Gibson, Toby J; Burwinkel, Barbara; Schneeweiss, Andreas; Pruneri, Giancarlo; Patil, Kiran R; Sotillo, Rocio; Jechlinger, Martin (2017) --- [The Journal of Clinical Investigation, 15 May 2017 ](https://doi.org/10.1172/JCI89914) --- [EMBL newspiece](https://news.embl.de/science/1705-cell-changes-drive-breast-cancer-relapse/) --- [Video Interview](https://youtu.be/nGNvYjYQaaQ)

- **Functional Outcomes and Quality of Life After Radical Prostatectomy Only Versus a Combination of Prostatectomy with Radiation and Hormonal Therapy.** -- Adam M, Tennstedt P, Lanwehr D, Tilki D, Steuber T, Beyer B, Thederan I, Heinzer H, Haese A, Salomon G, Budäus L, Michl U, Pehrke D, Stattin P, Bernard J, **Klaus B**, Pompe RS, Petersen C, Huland H, Graefen M, Schwarz R, Huber W, Loeb S, Schlomm T. (2017) ---
[Eur. Urol. 71(3):330-336](http://dx.doi.org/10.1136/gutjnl-2016-311477) 

- **miR-16 and miR-125b are involved in barrier function dysregulation through the modulation of claudin-2 and cingulin expression in the jejunum in IBS with diarrhoea.** --- Martínez C, Rodiño-Janeiro BK, Lobo B, Stanifer ML, **Klaus B**, Granzow M, González-Castro AM, Salvo-Romero E, Alonso-Cotoner C, Pigrau M, Roeth R, Rappold G, Huber W, González-Silos R, Lorenzo J, de Torres I, Azpiroz F, Boulant S, Vicario M, Niesler B, Santos J. (2017)  --- [Gut](http://dx.doi.org/10.1136/gutjnl-2016-311477) 

###  2016 


-  **Statistical relevance—relevant statistics, part II**
--- **Bernd Klaus**, 2016 --- [EMBO J](http://dx.doi.org/10.15252/embj.201694659) 

- **An end to end workflow for differential gene expression using Affymetrix microarrays**  --- **Bernd Klaus** --- version 1 , 2016 --- [F1000Research 2016, 5:1384](http://dx.doi.org/10.12688/f1000research.8967.1) 

- **Data-driven hypothesis weighting increases detection power in genome-scale multiple testing**  --- Nikolaos Ignatiadis, **Bernd Klaus**, Judith Zaugg, Wolfgang Huber, 2016 --- [Nature Methods](http://dx.doi.org/10.1038/nmeth.3885) --- earlier [BioRxiv Preprint](http://dx.doi.org/10.1101/034330) --- Software package [IHW on Github](https://github.com/nignatiadis/IHW) and [Bioconductor](http://bioconductor.org/packages/IHW).


- **DChIPRep, an R/Bioconductor package for differential enrichment analysis in chromatin studies**  --- Christophe D Chabbert, Lars M Steinmetz, **Bernd Klaus**, 2016  --- [PeerJ](https://peerj.com/articles/1981/) --- Software package [DChIPRep on Bioconductor](http://bioconductor.org/packages/DChIPRep). --- [Omic Tools website](https://omictools.com/dchiprep-tool) 


-  **Landscape and dynamics of transcription initiation in the malaria parasite Plasmodium falciparum**
--- Sophie Adjalley, Christophe Chabbert, **Bernd Klaus**, Vicent Pelechano, Lars Steinmetz, 2016.--- [Cell Reports, Volume 14, Issue 10, p2463–2475](http://dx.doi.org/10.1016/j.celrep.2016.02.025) --- earlier [BioRxiv Preprint](http://dx.doi.org/10.1101/024356)   --- [EMBL newspiece](http://news.embl.de/science/1602-malaria/).

- **CYP3A5 mediates basal and acquired therapy resistance in different subtypes of pancreatic ductal adenocarcinoma**  --- Elisa M Noll,  Christian Eisen,  Albrecht Stenzinger,  Elisa Espinet,  Alexander Muckenhuber,  Corinna Klein,  Vanessa Vogel,  **Bernd Klaus**,  Wiebke Nadler,  Christoph Rösli,  Christian Lutz,  Michael Kulke,  Jan Engelhardt,  Franziska M Zickgraf,  Octavio Espinosa,  Matthias Schlesner,  Xiaoqi Jiang,  Annette Kopp-Schneider,  Peter Neuhaus,  Marcus Bahra,  Bruno V Sinn,  Roland Eils,  Nathalia A Giese,  Thilo Hackert,  Oliver Strobel,  Jens Werner,  Markus W Büchler,  Wilko Weichert,  Andreas Trumpp  & Martin R Sprick, 2016  --- [Nature Medicine, 22, 278–287](https://dx.doi.org/10.1038/nm.4038).

###  2015 


-  **Statistical relevance—relevant statistics, part I**
--- **Bernd Klaus**, 2015 --- [EMBO J (2015) 34, 2727-2730](http://dx.doi.org/10.15252/embj.201592958) 


-  **A high-throughput ChIP Seq for large-scale chromatin studies**
--- Christophe D Chabbert, Sophie H Adjalley, **Bernd Klaus**, Emilie S Fritsch, Ishaan Gupta, Vicent Pelechano, Lars M Steinmetz, 2015.--- [Mol Syst Biol.; 11(1):777](http://www.dx.doi.org/10.15252/msb.20145776) --- [EMBL newspiece](http://news.embl.de/science/1501_barcoding-epigenetics/)

###  2014 


- **A Genome-Wide Map of Mitochondrial DNA Recombination in Yeast**
--- Fritsch, E.S., Chabbert, C.D., **Klaus, B**. & Steinmetz, L.M.,
2014  --- [Genetics](http://www.dx.doi.org/10.1534/genetics.114.166637) --- also
[issue highlight](http://www.genetics.org/content/198/2/NP) in Genetics 10/2014

- **Neural lineage induction reveals multi-scale dynamics of 3D chromatin organization**
 --- Aleksandra Pekowska, **Bernd Klaus**, Felix Alexander Klein, Simon Anders, Malgorzata Oles, Lars Michael Steinmetz, Paul Bertone, Wolfgang Huber, 2014. --- [BioRxiv Preprint](http://biorxiv.org/content/early/2014/04/11/004168)

-  **Alternative polyadenylation diversifies post-transcriptional regulation by selective RNA-protein interactions**
--- Gupta, I., Clauder-Munster, S., **Klaus, B**., Jarvelin, A.I., Aiyar, R.S., Benes, V., Wilkening, S., Huber, W., Pelechano, V. & Steinmetz, L.M., 2014.--- [Mol Syst Biol. 2014; 10(2):719](http://www.dx.doi.org/10.1002/msb.135068) --- [Research highlight in Nature Reviews Genetics](http://www.dx.doi.org/doi:10.1038/nrg3710)

###  2009 --- 2013

-  **Effect size estimation and misclassification rate based variable selection in linear discriminant analysis**
--- **B. Klaus**, 2013 --- [Journal of Data Science,v.11, no.3, p.559-573](http://www.jds-online.com/volume-11-number-3-july-2013) --- [arXiv:1205.6653](http://arxiv.org/abs/1205.6653)

- **Experimentally induced hyperchloremic and dl-lactic acidosis in calves: An attempt to study the effects of oral rehydration on acid-base status** ---  L. Schwedhelm, D. Kirchner, **B. Klaus**, L. Bachmann,  2013 --- [Journal of Dairy Science](http://dx.doi.org/10.3168/jds.2012-6077)


- **False Discovery Rates, Higher Criticism and Related Methods in High-Dimensional Multiple Testing** ---
**B. Klaus**, 2013 --- [PhD thesis](http://nbn-resolving.de/urn:nbn:de:bsz:15-qucosa-102676) --- 
[University of Leipzig](http://www.fmi.uni-leipzig.de/cms/en/forschung/promotionen.html)

- **Signal identification for rare and weak features: higher criticism 
or false discovery rates?** --- **B. Klaus** and K. Strimmer, 2013 ---
[Biostatistics](http://biostatistics.oxfordjournals.org/content/14/1/129)

- **Learning false discovery rates by fitting sigmoidal threshold functions** ---
**B. Klaus** and K. Strimmer. 2011 --- [Journal de la Soci&eacute;t&eacute;  Fran&ccedil;aise de Statistique](http://journal-sfds.fr/index.php/J-SFdS/issue/view/9) --- [arXiv:1104.5414](http://arxiv.org/abs/1104.5414)


- **Thresholding methods for feature selection in genomics: higher criticism versus false non-discovery rates** ---
**B. Klaus** and K. Strimmer. 2010 ---
Proceedings of the [7th International Workshop on Computational Systems  Biology](http://www.cs.tut.fi/wcsb10/), WCSB 2010
(June 16-18, 2010, Luxembourg, Luxembourg) --- [reprint](http://strimmerlab.org/publications/conferences/hc-comparison2010.pdf)


- **Ueber eine mit dem Stieltjesschen Potenzmomentenproblem im Zusammenhang stehende Klasse holomorpher Matrixfunktionen**. (On a class of [holomorphic](http://en.wikipedia.org/wiki/Holomorphic_function) matrix---valued functions and
their connection to the [Stieltjes](http://en.wikipedia.org/wiki/Thomas_Joannes_Stieltjes) [moment problem](http://en.wikipedia.org/wiki/Moment_problem), in German) [pdf](http://www-huber.embl.de/users/klaus/Diplomarbeit_Bernd_Klaus.pdf), Diploma thesis (equivalent to M. Sc.), [Institute  of Mathematics](http://www.math.uni-leipzig.de/cms/en/startseite/), [University of Leipzig](http://www.zv.uni-leipzig.de/en/), 2009


## Software

### Chromium

Chromium is a work--in--progress R package for the comprehensive analysis of Hi--C
data developed by [Annika Gable](https://www.imls.uzh.ch/en/research/vonmering/people/annika-gable.html), [Aleksandra Pekowska](http://casellaslab.github.io/People/) and myself. You can find it [on github](https://github.com/AnnikaGable/chromium).

### DChIPRep

The  [DChIPRep](http://bioconductor.org/packages/DChIPRep/) DChIPRep 
implements a methodology to assess differences between chromatin modification profiles in replicated ChIP-Seq studies as described in [Chabbert et. al.](http://www.dx.doi.org/10.15252/msb.20145776) and is available from
[Bioconductor](http://bioconductor.org/packages/DChIPRep/).

### PhD Thesis

* Script files related to my PhD thesis can be downloaded [here](http://www-huber.embl.de/users/klaus/software/PhD-BK-Soft.zip).

* Script files implementing the effect sizes variable selection methods presented in 
the article "Effect Size Estimation and Misclassification Rate Based Variable 
Selection in Linear Discriminant Analysis" can be downloaded  [here](http://www-huber.embl.de/users/klaus/software/LDA-Effect.zip).



### concaveFDR

concaveFDR is a an FDR estimation package, similar in spirit to [fdrtool](http://strimmerlab.org/software/fdrtool/index.html)
and [locfdr](http://cran.r-project.org/package=locfdr). It will be submitted to Bioconductor at some point and extensive 
documentation will be become available.

- The R package (beta status!) via github like so:

```{r , eval=FALSE}
devtools::install_github("b-klaus/concaveFDR")
```


You can download some introductory [slides from MAS Toulouse 2014](http://www-huber.embl.de/users/klaus/270814-tolouse-MAS-FDR-Testing-klaus.pdf). Most of the  methodology is also treated in my [PhD thesis](http://nbn-resolving.de/urn:nbn:de:bsz:15-qucosa-102676).



#### Simulation Studies and a re-analysis of exon usage conservation

There are two simulation studies and one real data analysis of exon usage conservation.
For the background on the exon usage analysis see the original paper: [Reyes et. al. --- Drift and conservation of differential exon usage across tissues in primate species](http://dx.doi.org/10.1073/pnas.1307202110). All of the files
can obtained via github: <https://github.com/b-klaus/data_analysis_exon_usage>

  * Simulation Study with  normal mixture models: [Rmd](https://github.com/b-klaus/data_analysis_exon_usage/tree/master/Sims/SimsZvals.Rmd)
  
  * Simulation Study with  the half normal decay model: [Rmd](https://github.com/b-klaus/data_analysis_exon_usage/tree/master/Sims/SimsHND-Model.Rmd)
    
  * Exon Usage Analysis: [Rmd](https://github.com/b-klaus/data_analysis_exon_usage/tree/master/AnalysisAlejandro.Rmd)

